File: y.js
Y = f =>
    (x => f(y => x(x)(y)))
    (x => f(y => x(x)(y)))

fact = Y(g => n => n == 0 ? 1 : g(n - 1) * n)
fib = Y(g => n => n <= 2 ? 1 : g(n - 1) + g(n - 2))

console.log(fact(10));
console.log(fib(10));

File: y.py
Y = (lambda fn: (lambda f: f(f))(lambda f: fn(lambda n: f(f)(n))))
fib = Y(lambda g: lambda n: 1 if n in [1, 2] else g(n-1) + g(n-2))
fact = Y(lambda g: lambda n: 1 if n == 0 else g(n-1)*n)

print(fib(10))
print(fact(10))
